examples/autoignition-mpi.py

```diff
@@ -499,25 +499,65 @@ def my_get_timestep(t, dt, state):
                                 op="max")
         return ts_field, cfl, min(t_remaining, dt)
 
-    def limiter(cv, temp=None):
+    def _limit_fluid_state(fluid_state):
+        pressure = fluid_state.dv.pressure
+        temperature = fluid_state.dv.temperature
+        cv = fluid_state.cv
+
+        # limit species
         spec_lim = make_obj_array([
-            bound_preserving_limiter(dcoll, cv.species_mass_fractions[i], mmax=1.0)
+            bound_preserving_limiter(dcoll, cv.species_mass_fractions[i], mmin=0.0)
             for i in range(nspecies)
         ])
 
-        kin_energy = 0.5*np.dot(cv.velocity, cv.velocity)
+        # normalize to ensure sum_Yi = 1.0
+        aux = cv.mass*0.0
+        for i in range(0, nspecies):
+            aux = aux + spec_lim[i]
+        spec_lim = spec_lim/aux
+
+        # recompute density
+        mass_lim = eos.get_density(pressure=pressure,
+            temperature=temperature, species_mass_fractions=spec_lim)
 
-        energy_lim = cv.mass*(
-            gas_model.eos.get_internal_energy(temp, species_mass_fractions=spec_lim)
-            + kin_energy
+        # recompute energy
+        energy_lim = mass_lim*(gas_model.eos.get_internal_energy(
+            temperature, species_mass_fractions=spec_lim)
+            + 0.5*np.dot(cv.velocity, cv.velocity)
         )
 
-        return make_conserved(dim=dim, mass=cv.mass, energy=energy_lim,
-                       momentum=cv.momentum, species_mass=cv.mass*spec_lim)
+        # make a new CV with the limited variables
+        limited_cv = make_conserved(dim=dim, mass=mass_lim, energy=energy_lim,
+            momentum=mass_lim*cv.velocity, species_mass=mass_lim*spec_lim)
+
+        # since temperature nor pressure are changed during the limiting process,
+        # it is possible to avoid recomputing it by NOT calling make_fluid_state.
+        # Thus, evaluate ONLY other DV and TV variables for the limited state
+        new_dv = MixtureDependentVars(
+            temperature=temperature,
+            pressure=pressure,
+            speed_of_sound=eos.sound_speed(limited_cv, temperature),
+            species_enthalpies=eos.species_enthalpies(limited_cv, temperature))
```

Comment:
Is there a performance reason not to use this:
```suggestion
        new_dv = fluid_state.dv.replace(
               speed_of_sound=eos.sound_speed(limited_cv, temperature)
               species_enthalpies=eos.species_enthalpies(limited_cv, temperature)
        )
```

Reply:
Not that I am aware. Changing to your suggestion.